application/templates/secretProviderClass.yaml

```diff
@@ -0,0 +1,23 @@
+{{- if .Values.secretProviderClass.enabled }}
+apiVersion: secrets-store.csi.x-k8s.io/v1alpha1
+kind: SecretProviderClass
+metadata:
+  name: {{.Values.secretProviderClass.name}}
+spec:
+  provider: {{.Values.secretProviderClass.provider}}
+  parameters:
+    vaultAddress: {{.Values.secretProviderClass.parameters.vaultAddress}}
+    roleName: {{.Values.secretProviderClass.parameters.roleName}}
```

Comment:
I would say remove take this value from .Values.secretProviderClass.vaultAddress

lets not use .parameter

Reply:
Done